Homework 2

LukaVukovic2 opened the pull request:
Migrated Homework 1 assignments to React. Improved composability and structure by dividing code into multiple components while preserving previous functionalities and logic.

LukaVukovic2 committed:
Migrate project to Next.js

Successfully migrated data and structured it according to the rules.
Default review list data, which is being passed as props, renders on
page. Created new interfaces for shows and review(s). Meta data are
stored in separate component. Each component has its own CSS file.

LukaVukovic2 committed:
Migrate entire project to Next.js

1. assignment
Complete logic of first two assignments is preserved. Conditional
rendering is applied - message is
displayed if there are no reviews. Chakra UI is used
for built in components. Nanoid library helps with identifying each
review.

LukaVukovic2 committed:
Code restructure

Adjusted components name and directory position. Reviewer avatar and
email is rendered next to review details.

LukaVukovic2 committed:
Implement rating input component using star icons

Rating can be selected by click on star which colors all previous stars.
Structure and composability is improved with separating rating input logic into
StarIcon component.

LukaVukovic2 committed:
Code cleaning

Removed unnecessary elements and CSS classes.

ante-pesorda commented on homework-2/src/app/globals.css:
```diff
@@ -0,0 +1,15 @@
+body{
+  margin: 0;
+  padding: 0 20px 20px 20px;
+  background-color: navy !important;
```
!important smo rekli što više izbjegavati. Bolje zadati jači specificity

ante-pesorda commented on homework-2/src/components/features/shows/ReviewForm/ReviewForm.tsx:
```diff
@@ -0,0 +1,99 @@
+  }
+
+  function resetFormInputs() {
+    const stars = document.querySelectorAll(
+      ".star"
+    ) as NodeListOf<HTMLElement>;
```
možda pokušat riješit koristeći useRef hook

ante-pesorda commented on homework-2/src/components/shared/MainLayout/MainLayout.tsx:
```diff
@@ -0,0 +1,86 @@
+    setReviewArr(arr);
+  }, []);
+
+  useEffect(() => {
+    calculateAverageRating(reviewArr);
+  }, [reviewArr]);
```
bi li mogli kako bez ovog useEffect-a?

LukaVukovic2 committed:
Code optimization

Replaced DOM selectors with 'useRef' hooks. Removed
unnecessary !important CSS properties. Reduced useEffect hook usage.

ante-pesorda commented on homework-2/src/components/features/shows/ReviewForm/ReviewForm.tsx:
```diff
@@ -0,0 +1,106 @@
+      avatar:
+        "https://st3.depositphotos.com/6672868/13701/v/380/depositphotos_137014128-stock-illustration-user-profile-icon.jpg",
+      email: "email@example.com",
+      comment: comment,
+      rating: rating,
```
Kad imamo key i value parametar iste po nazivu, možemo ovako kraće pisat
```suggestion
      comment,
      rating,
```

ante-pesorda reviewed (approved):
🔥
